src/Configurator.php

```diff
@@ -169,6 +174,19 @@ public function withSetting($name, $value)
     }
 
     /**
+     * @param $extension string
+     * @param $className string
+     *
+     * @return Configurator
```

Comment:
Can we use `$this` here to indicate that the same instance is returned, instead of a new instance?